src/main/java/mate/academy/internetshop/dao/jdbc/BasketDaoJdbcImpl.java

```diff
@@ -0,0 +1,217 @@
+                basketId = rs.getLong("basket_id");
+                userIdfromBd = rs.getLong("user_id");
+                itemId = rs.getLong("item_id");
+                Item item = itemDao.get(itemId).get();
+                items.add(item);
+            }
+            Basket basket = new Basket(basketId, userIdfromBd, items);
+            logger.info("Get all optional baskets from db");
+            return Optional.of(basket);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all items from bucket by user id", e);
+        }
+    }
+
+    @Override
+    public Basket create(Basket basket) {
+        String sql = "INSERT INTO shop.baskets (user_id) VALUES (?);";
+
+        try (Connection connection = DbConnector.connect();
+             PreparedStatement statement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, basket.getUserId());
+            statement.executeUpdate();
+            ResultSet rs = statement.getGeneratedKeys();
+            while (rs.next()) {
+                Long basketId = rs.getLong(1);
+                basket.setId(basketId);
+            }
+            addItems(basket, basket.getItems());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create basket ", e);
+        }
+        return basket;
+    }
+
+    private void addItems(Basket basket, List<Item> items) {
+        String sql = "INSERT INTO shop.items_baskets(basket_id, item_id) VALUES (?, ?);";
+        changeBasketItems(basket, items, sql);
+    }
+
+    private void changeBasketItems(Basket basket, List<Item> items, String sql) {
+        try (Connection connection = DbConnector.connect();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            for (Item item : items) {
+                statement.setLong(1, basket.getId());
+                statement.setLong(2, item.getId());
+                statement.executeUpdate();
+            }
+            logger.info("add items in baskets");
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't changed in basket list items ", e);
+        }
+    }
+
+    @Override
+    public Optional<Basket> get(Long id) {
+        String sql = "SELECT * FROM shop.baskets INNER JOIN shop.items_baskets ON baskets.basket_id = items_baskets.basket_id " +
+                "WHERE baskets.basket_id = ?;";
+        try (Connection connection = DbConnector.connect();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+            statement.setLong(1, id);
+            ResultSet rs = statement.executeQuery();
+
+            Long basketId = null, userId = null;
+            List<Long> items = new ArrayList<>();
+
+            while (rs.next()) {
+                basketId = rs.getLong("basket_id");
+                userId = rs.getLong("user_id");
+                items.add(rs.getLong("item_id"));
```

Comment:
```suggestion
               itemsFromBd.add(itemDao.get(rs.getLong("item_id").get());
```
і таким чином нам не буде потрібний List<Long> items = new ArrayList<>();